### PR TITLE
feat!: convert library to TypeScript, adding v1p1beta1 import

### PR DESCRIPTION
**Pull request by bcoe**
This moves us to the TypeScript generator, which allows us to start pulling in fancy :jeans: new updates to the API.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,34 +40,32 @@
...
   "dependencies": {
-    "google-gax": "^1.7.5",
-    "protobufjs": "^6.8.0"
+    "google-gax": "^1.7.5"
   },
   "devDependencies": {
+    "@types/mocha": "^5.2.5",
+    "@types/node": "^12.0.0",
+    "c8": "^7.0.0",
     "codecov": "^3.0.0",
```
Comment (xiaozhenliu-gg5) [inline package.json]: `codecov` is not in use now I think, maybe we can safely remove this?

File: webpack.config.js [hunk 2]
```diff
@@ -12,10 +12,12 @@
...
 // limitations under the License.
 
+const path = require('path');
+
 module.exports = {
-  entry: './src/browser.js',
+  entry: './src/index.ts',
   output: {
-    library: 'security-center',
+    library: 'SecurityCenter',
```
Comment (xiaozhenliu-gg5) [inline webpack.config.js]: Not sure if it's a breaking change for library name @alexander-fenster ?

**Comment by alexander-fenster** [comment]
This will be breaking because the set of path templates is different (we are missing some helper methods such as `assetSecurityMarksPath(organization, asset)`). This is not a problem with your generation, it's an annotation problem that I'm trying to figure out with my team.

We can either merge it as a breaking change now (and if annotations are fixed later, add them), or wait for a while until this is sorted out. I will let you know!

File: synth.py [hunk 3]
```diff
@@ -20,65 +20,47 @@
...
-        'https:\/\/cloud\.google\.com[\s\*]*http:\/\/(.*)[\s\*]*\)',
-        r"https://\1)")
-
-s.replace('**/doc/google/protobuf/doc_timestamp.js',
-        'toISOString\]',
-        'toISOString)')
-# [END fix-dead-link]
+# There's an internal tracking ticket for this generation bug b/148673437
+# we should remove this replacement hack as soon as it is addressed:
+s.replace("protos/google/cloud/securitycenter/v1p1beta1/notification_config.proto", r"""// The Pub/Sub Topic resource definition is in google/cloud/pubsub/v1/,
```
Comment (alexander-fenster) [inline synth.py]: This is cool! I'm making this change internally, then we'll be able to remove this hack. Internal CL 292600161

**Comment by alexander-fenster** [comment]
https://github.com/googleapis/gapic-generator-typescript/issues/220 will take care of this.

File: package.json [hunk 4]
```diff
@@ -1,18 +1,31 @@
 {
   "name": "@google-cloud/security-center",
   "description": "Cloud Security Command Center API client for Node.js",
-  "version": "2.3.2",
+  "version": "2.3.1",
```
Comment (JustinBeckwith) [inline package.json]: Huh, the version drop is weird

**Comment by xiaozhenliu-gg5** [comment]
Newest image is published and I generated `security-center` which picks the missing resources back, can you try one more time? @bcoe Thanks!

**Pull request merged**